[WFLY-17743]: Refactor embedded-activemq and remote-activemq Galleon layers.

ehsavoie opened the pull request:
* embedded-activemq layer now depends on messaging-activemq layer instead of resource-adapters.
 * remote-activemq layer now depends on messaging-activemq layer instead of resource-adapters.

Jira: https://issues.redhat.com/browse/WFLY-17743

bstansberry reviewed (changes requested):
@ehsavoie Please include an update to https://github.com/wildfly/wildfly/blob/main/docs/src/main/asciidoc/_galleon/Galleon_layers.adoc

ehsavoie committed:
[WFLY-17743]: Refactor embedded-activemq and remote-activemq Galleon layers.

 * embedded-activemq layer now depends on messaging-activemq layer
   instead of resource-adapters.
 * remote-activemq layer now depends on messaging-activemq layer
instead of resource-adapters.

Jira: https://issues.redhat.com/browse/WFLY-17743

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

ehsavoie commented:
@bstansberry done

bstansberry commented:
Thanks @ehsavoie
